src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -134,7 +134,7 @@ export interface DiscoverStateContainer {
      * Used by the Data View Picker
      * @param pattern
      */
-    onCreateDefaultAdHocDataView: (dataViewSpec: DataViewSpec) => Promise<void>;
+    onCreateDefaultAdHocDataView: (dataViewSpec: DataViewSpec) => Promise<DataView>;
```

Comment:
As this method creates a new ad-hoc data view and stores it, I didn't interpret it as an event handler.
I also needed to retrieve the instance of the newly created DataView when applying the customizations to set this DataView directly during the customization.

Could this be renamed something like `createAndAppendAdHocDataView`?

Reply:
Yup, renaming to `createAndAppendAdHocDataView` makes sense to me.